Library/Homebrew/os/linux.rb

```diff
@@ -25,6 +27,23 @@ def os_version
         "Unknown"
       end
     end
+
+    sig { returns(T::Boolean) }
+    def wsl?
+      /-microsoft/i.match?(OS.kernel_version.to_s)
+    end
+
+    sig { returns(Version) }
+    def wsl_version
+      Version::NULL unless wsl?
+      kernel = OS.kernel_version.to_s
+      return Version.new("2 (Microsoft Store)") if Version.new(T.must(kernel[/^([0-9.]*)-.*/,
+                                                                             1])) > Version.new("5.15")
+      return Version.new("2") if kernel.include?("-microsoft")
+      return Version.new("1") if kernel.include?("-Microsoft")
+
+      Version::NULL
```

Comment:
```suggestion
      if Version.new(T.must(kernel[/^([0-9.]*)-.*/, 1])) > Version.new("5.15")
        Version.new("2 (Microsoft Store)")
      elsif kernel.include?("-microsoft")
        Version.new("2")
      elsif kernel.include?("-Microsoft")
        Version.new("1") 
      else
        Version::NULL
      end
```

Reply:
Are you sure this `T.must` is the best way to handle this/correct? It feels like you're asserting on a value that may not be guaranteed to be non-nil, right?

Reply:
T.must was added by `brew typecheck`. `uname -r` reports Kernel's release, so I guess `T.must` should be safe to use, but 🤷🏻